configsuite/schema.py

```diff
@@ -76,9 +76,17 @@ def _check_required_not_default(schema_level):
     return True
 
 
+@configsuite.validator_msg("Only variable lenght containers can specify allowempty")
+def _check_allowempty_only_variable_containers(schema_level):
+    level_type = schema_level.get(MK.Type)
+    const_len = isinstance(level_type, types.BasicType) or level_type == types.NamedDict
```

Comment:
What does `const_len` mean? And why vary between `isinstance` and `==` for comparator? A comment would maybe clarify that.

Reply:
As the name of the function hints towards, we now distinguish between variable length containers (list and dict) and constant length data types (basic type and named dict). The idea is that only the former is allowed to specify `allowempty`, as the notion has no meaning for the later.

Reply:
I agree that the mixture between `isinstance` and `==` is unfortunate. It is due to the fact that `BasicType` is a meta type in configsuite (and is implemented as a class), while `NamedDict` is a concrete type (implemented as an instance of `Container`). This is an implementation detail that is leaking too far and we could very well abstract the difference in implementation away in a meaningful manner!

Reply:
My hope was that the validation message, the function name and the variable name (which should perhaps have been `is_const_length`?) would be sufficient documentation. But I'm more than willing to add a description as well :) If you mind my answers above clarifying; do you have any suggestions for a helpful comment?

Reply:
No, this becomes clearer after reading the docs/loading the context of the PR.